src/modules/commander/state_machine_helper.cpp

```diff
@@ -330,7 +330,7 @@ transition_result_t arming_state_transition(vehicle_status_s *status,
 				}
 
 				feedback_provided = true;
-				valid_transition = false;
+				valid_transition = true;
```

Comment:
Preflight check during transition to ARMING_STATE_STANDBY shouldn't cause the state transition to fail.

Reply:
Why not, isn't that its purpose?

ARMING_STATE_STANDBY = preflight checks passed?

Reply:
Hi @dagar, @mhkabir 
Sorry for the late reply, I'm out of country and do not have access to mail.

When I added this change I was facing some timing issue related to arming_state_transition.
The timing issue stayed hidden when the arming_state_transition is called from the low_prio_thread. As the calibration was moved to the the main thread, it was retuning TRANSITION_DENIED for the following.
https://github.com/PX4/Firmware/pull/8706/files#diff-1d51a813dcb26f46616f71f28aacc70dR1079

I already explained about the timing issue that was observed earlier in the following comment, when I made this change.
https://github.com/PX4/Firmware/pull/8706#issuecomment-361893470

I see recently there has been many fixes in state_machine_helper.cpp, I think removing this condition might work now. I can check on the actual hardware only after I return in few days.